### PR TITLE
Honor initial_layout when coupling map is None

### LINKED ISSUE #5345: transpile(..., initial_layout=...) is ignored if there is no coupling map
### Information
 
- **Qiskit Terra version**: master
- **Python version**: all
- **Operating system**: all

### What is the current behavior?

```python
from qiskit import QuantumCircuit
from qiskit.compiler import transpile
from qiskit.transpiler import CouplingMap

circ = QuantumCircuit(2)
circ.h(0)
circ.cx(0,1)

transpile(circ, routing_method='basic', initial_layout=[1, 0]).draw()
```
```
     ┌───┐     
q_0: ┤ H ├──■──
     └───┘┌─┴─┐
q_1: ─────┤ X ├
          └───┘
```

Notice that, when a coupling map is provided, the output is:
```python
transpile(circ, routing_method='basic', initial_layout=[1, 0], coupling_map=[[1,0], [0,1]]).draw()
```
```
              ┌───┐
q_1 -> 0 ─────┤ X ├
         ┌───┐└─┬─┘
q_0 -> 1 ┤ H ├──■──
         └───┘     
```
### What is the expected behavior?

`transpile` should apply the layout, even when there is no coupling map.

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #5345 

This PR embeds the layout when `initial_layout` is provided, even when `coupling_map=None`.

Additionally, it allows `coupling_map=None` in `FullAncillaAllocation` and `CheckMap`

### CODE REVIEW

File: test/python/transpiler/test_preset_passmanagers.py [hunk 1]
```diff
@@ -167,6 +167,40 @@ def test_symmetric_coupling_map(self, level):
...
+        """Honor initial_layout (partially defined) when coupling_map=None
+        See: https://github.com/Qiskit/qiskit-terra/issues/5345
+        """
+        qr = QuantumRegister(2, 'q')
+        qc = QuantumCircuit(qr)
+        qc.h(qr[0])
+        qc.cx(qr[0], qr[1])
+
+        transpiled = transpile(qc, initial_layout=[4, 2], optimization_level=level,
+                               callback=self.callback)
```
Comment (ajavadia) [inline test/python/transpiler/test_preset_passmanagers.py:198]: so what does `transpiled` look like here? is it a 2 qubit circuit or a 5 qubit circuit?

**Comment by ajavadia** [comment]
can you add a release note and a note to the `initial_layout` arg of `transpile()` which says if no coupling_map is given, then this just assigns the virtual qubits to the respective physical qubits.

File: test/python/transpiler/test_preset_passmanagers.py (hunk 1 above)
Comment (1ucian0) [reply test/python/transpiler/test_preset_passmanagers.py:198]: 5, see line 202

**Comment by 1ucian0** [comment]
Release note added in 0a552752f

File: test/python/transpiler/test_preset_passmanagers.py (hunk 1 above)
Comment (chriseclectic) [reply test/python/transpiler/test_preset_passmanagers.py:198]: I think taking the max of the initial layout makes sense for these situations.

**Review by kdk — APPROVED** [review]

**Pull request merged**